### PR TITLE
WNC new platfrom osw1800

### PR DESCRIPTION
**Pull request by YaoTien**
Upload for WNC new platform, osw1800.

### CODE REVIEW

**Comment by krambn** [comment]
Looks good. Couple of questions - on what system do you build the debian packages? Can we not change the bfn-sai.mk? Looks like the dependency of the build machine is forcing you to make that change.

**Comment by YaoTien** [comment]
1. I generated debian packages from ubuntu system.
2. Barefoot split their SDK as two parts, bfn-sdk and bfn-platform.
bfn-sdk includes common driver, SAI and all switch protocols. So Barefoot prohibit vendors to change it.
bfn-platform is opened vendors do something change, as like port's mapping table, LED definition and some vendor special functions. So Barefoot only allow vendors to change this.

**Comment by krambn** [comment]
Can you build on a debian 8 based system ? That way the dependent libraries can match up? The version and type of system used to build could force library dependencies that are incompatible between the versions we load onto the sonic build - which is based on debian8.

**Comment by YaoTien** [comment]
Sorry for my wrong information. I built the debian packages in debian 8 system. And we also built SONiC image and executed on our device. It worked well.

**Comment by krambn** [comment]
Thanks. I will try to build to include your platform and try to merge the pull request.

File: platform/barefoot/bfn-platform-wnc.mk [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BFN_PLATFORM = bfnplatformwnc_1.0.0_amd64.deb
```
Comment (krambn) [inline platform/barefoot/bfn-platform-wnc.mk]: should be BFN_WNC_PLATFORM

File: platform/barefoot/bfn-platform-wnc.mk [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+$(BFN_PLATFORM)_URL = "https://github.com/YaoTien/download/raw/master/sonic/sde/7_0_0_18/bfnplatformwnc_1.0.0_amd64.deb"
```
Comment (krambn) [inline platform/barefoot/bfn-platform-wnc.mk]: $(BFN_WNC_PLATFORM)_URL

**Comment by krambn** [comment]
Binary built with wrong package name: https://github.com/YaoTien/download/raw/master/sonic/sde/7_0_0_18/bfnplatformwnc_1.0.0_amd64.deb

This debian package should be built with correct name - something like 

**dh_make --native --single --packagename bfnplatformwnc_1.0.0 --email support@wncnetworks.com -c apache**

Currently built **incorrectly** with bfnplatform_1.0.0

Note: The names of the platforms are expected to be unique to prevent collisions in package names!

**Comment by YaoTien** [comment]
Sorry for my mistakes.
I updated platform name and re-built it successfully.
Please update code and try it.
Thanks.

**Comment by krambn** [comment]
Can you please try the command and verify the package name is right?
I tried with this file: https://github.com/YaoTien/download/raw/master/sonic/sde/7_0_0_18/bfnplatformwnc_1.0.0_amd64.deb

dpkg --info bfnplatformwnc_1.0.0_amd64.deb
 new debian package, version 2.0.
 size 311760 bytes: control archive=1072 bytes.
     459 bytes,    11 lines      control
     759 bytes,     8 lines      md5sums
     135 bytes,     7 lines   *  postinst             #!/bin/sh
     132 bytes,     7 lines   *  postrm               #!/bin/sh
     146 bytes,     5 lines      shlibs
 Package: **bfnplatform**
 Version: 1.0.0
 Architecture: amd64

Package should be **bfnplatformwnc**

Thanks

**Comment by YaoTien** [comment]
I updated the file. Please try it again.
Thanks.

**Comment by krambn** [comment]
Looks good. Will merge soon. Thanks

**Pull request merged**